src/Repository.ts

```diff
@@ -248,9 +248,11 @@ export class MongoRepository<DOC, DTO = DOC> {
   async deleteOne(conditions: any): Promise<DeleteWriteOpResultObject> {
     const collection = await this.collection;
 
-    await this.invokeEvents(PRE_KEY, [RepoOperation.delete, RepoOperation.deleteOne], conditions);
+    let document = await collection.findOne(conditions);
+    document = this.toggleId(document, false);
+    await this.invokeEvents(PRE_KEY, [RepoOperation.delete, RepoOperation.deleteOne], null, document);
```

Comment:
should we be short circuiting when the document is `null` (not found)? would imagine the events not be invoked at least.

Reply:
Good idea